hamilton/lifecycle/default.py

```diff
@@ -376,7 +377,10 @@ def run_to_execute_node(
         if node_name not in self.cache_vars:
             return node_callable(**node_kwargs)
 
-        node_hash = graph_types.hash_source_code(node_callable, strip=True)
+        source_of_node_callable = node_callable
+        while isinstance(source_of_node_callable, partial):  # handle partials
+            source_of_node_callable = source_of_node_callable.func
+        node_hash = graph_types.hash_source_code(source_of_node_callable, strip=True)
```

Comment:
instead of this code, you should use `node.version` (`HamiltonNode.version`). It uses the same code implementation, but the `CacheAdapter` predates the `.version`

Reply:
that's not wired through.

Reply:
Then, it should be retrieved from `graph: HamiltonGraph` in `run_before_graph_execution`

Reply:
We shoudln't just wire it through for the sake of  using the same function. But yeah, that brings up a point of why are we computing this here when we compute statically? We should just compute all the hash keys in advance.

Let's just

Reply:
I think this is fine what I have now -- it's stop gap measure until new caching stuff comes out anyway.

Reply:
Yep, approved, ship it